webpack.config.js

```diff
@@ -6,89 +6,55 @@ const { CleanWebpackPlugin } = require('clean-webpack-plugin');
 const HOST = process.env.HOST || '0.0.0.0';
 const PORT = process.env.PORT || 3000;
 
-module.exports = [
-  {
-    name: 'dev',
-    entry: './src/index.js',
-    mode: 'development',
-    module: {
-      rules: [
-        {
-          test: /\.(js|jsx)$/,
-          exclude: /node_modules/,
-          use: ['babel-loader'],
-        },
-        {
-          test: /\.s[ac]ss$/i,
-          use: ['style-loader', 'css-loader', 'sass-loader'],
-        },
-      ],
-    },
-    resolve: {
-      extensions: ['*', '.js', '.jsx'],
-      alias: {
-        '@': path.join(__dirname, 'src'),
+const prodConfig = {
+  name: 'prod',
+  entry: './src/index.js',
+  mode: 'production',
+  module: {
+    rules: [
+      {
+        test: /\.(js|jsx)$/,
+        exclude: /node_modules/,
+        use: ['babel-loader'],
+      },
+      {
+        test: /\.s[ac]ss$/i,
+        use: ['style-loader', 'css-loader', 'sass-loader'],
       },
-    },
-    output: {
-      path: path.join(__dirname, 'dist'),
-      publicPath: './',
-      filename: 'bundle.js',
-    },
-    plugins: [
-      new CleanWebpackPlugin(),
-      new HtmlWebpackPlugin({
-        title: 'Basic React App',
-        template: './src/index.html',
-      }),
-      new webpack.HotModuleReplacementPlugin(),
-      new webpack.EnvironmentPlugin({API_URL: 'http://localhost:5000/api'}),
     ],
-    devServer: {
-      contentBase: './dist',
-      hot: true,
-      historyApiFallback: true,
-      writeToDisk: true,
-      host: HOST,
-      port: PORT,
-    },
   },
-  {
-    name: 'prod',
-    entry: './src/index.js',
-    mode: 'production',
-    module: {
-      rules: [
-        {
-          test: /\.(js|jsx)$/,
-          exclude: /node_modules/,
-          use: ['babel-loader'],
-        },
-        {
-          test: /\.s[ac]ss$/i,
-          use: ['style-loader', 'css-loader', 'sass-loader'],
-        },
-      ],
-    },
-    resolve: {
-      extensions: ['*', '.js', '.jsx'],
-      alias: {
-        '@': path.join(__dirname, 'src'),
-      },
-    },
-    output: {
-      path: path.join(__dirname, 'dist'),
-      publicPath: './',
-      filename: 'bundle.js',
+  resolve: {
+    extensions: ['*', '.js', '.jsx'],
+    alias: {
+      '@': path.join(__dirname, 'src'),
     },
-    plugins: [
-      new CleanWebpackPlugin(),
-      new HtmlWebpackPlugin({
-        // TODO: Add cache-buster (if needed)
-        title: 'Basic React App',
-        template: './src/index.html',
-      }),
-      new webpack.EnvironmentPlugin({API_URL: 'http://localhost:5000/api'}),
-    ],
   },
-];
+  output: {
+    path: path.join(__dirname, 'dist'),
+    publicPath: './',
+    filename: 'bundle.js',
+  },
+  plugins: [
+    new CleanWebpackPlugin(),
+    new HtmlWebpackPlugin({
+      title: 'Basic React App',
+      template: './src/index.html',
+    }),
+    new webpack.EnvironmentPlugin({ API_URL: 'http://localhost:5000/api' }),
+  ],
+};
+const devConfig = {
+  ...prodConfig,
```

Comment:
Is this intentional? I don't think it should be spread here since we're exporting it at the bottom

Reply:
Will discuss this further with @ashtonmeuser